packages/http_async/http_async.0.1.0/opam

```diff
@@ -0,0 +1,45 @@
+opam-version: "2.0"
+synopsis: "Async library for HTTP/1.1 servers"
+description:
+  "http_async implements an efficient HTTP/1.1 server. It uses the shuttle library for network IO, and provides an easy-to-use interface for writing async http services while minimal overhead on top of the user provided http handler."
+maintainer: ["Anurag Soni <anurag@sonianurag.com>"]
+authors: ["Anurag Soni <anurag@sonianurag.com>"]
+license: "MIT"
+tags: ["http-server" "http" "http1.1" "async"]
+homepage: "https://github.com/anuragsoni/http_async"
+doc: "https://anuragsoni.github.io/http_async/"
+bug-reports: "https://github.com/anuragsoni/http_async/issues"
+depends: [
+  "dune" {>= "2.9"}
+  "ocaml" {>= "4.11.0"}
+  "shuttle" {>= "0.6.0"}
+  "ppxlib" {>= "0.23.0"}
+  "odoc" {with-doc}
+]
+build: [
+  ["dune" "subst"] {dev}
+  [
+    "dune"
+    "build"
+    "-p"
+    name
+    "-j"
+    jobs
+    "--promote-install-files=false"
+    "@install"
+    "@runtest" {with-test}
+    "@doc" {with-doc}
+  ]
+  ["dune" "install" "-p" name "--create-install-files" name]
+]
+dev-repo: "git+https://github.com/anuragsoni/http_async.git"
+available: [ arch = "x86_64" | arch = "arm64" ]
```

Comment:
is there a particular reason for this restriction?

Reply:
In an older submission to the opam-repository there were some test failures on `s390x`. I haven't had a chance to spend time debugging those yet so I'm locking the architectures to the platform I have an easy way to use and test on. as for only supporting 64 bit archictures, I made some assumptions to represent http body lengths as an `int` instead of `int64` that will simplifies things on 64 bit systems, but will cause unintended issues on 32 bit archiectures.